File: gestion_cinema/src/main/java/net/yacine/gestion_cinema/entities/FilmScreening.java
package net.yacine.gestion_cinema.entities;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Builder;

@Entity @NoArgsConstructor @AllArgsConstructor @Data @Builder
public class FilmScreening {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate startDate;
    private int price;
    @ManyToOne
    private Film film;
    @ManyToOne
    private Room room;
    @ManyToOne
    private SessionFilmScreening sessionFilmScreening;
    @OneToMany(mappedBy = "filmScreening")
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<Ticket> tickets;
    
}

File: front_end_gestion_cinema/lib/data/models/filmScreening.dart
import 'package:front_end_gestion_cinema/data/models/film.dart';
import 'package:front_end_gestion_cinema/data/models/room.dart';
import 'package:front_end_gestion_cinema/data/models/session_film_screening.dart';

class FilmScreening {
  late String startDate;
  late int price;
  late String urlFilm;
  late String urlTickets;
  late String urlSessionFilmScreening;
  Room room = Room();
  Film film = Film();
  SessionFilmScreening sessionFilmScreening = SessionFilmScreening();

  FilmScreening(){}

  FilmScreening.fromJson(Map<String, dynamic> json) {
    startDate = json['startDate'];
    price = json['price'];
    urlFilm = json['_links']['film']['href'];
    urlTickets = json['_links']['tickets']['href'];
    urlSessionFilmScreening = json['_links']['sessionFilmScreening']['href'];
  }
}
